### DISCUSSION TITLE
Problème de précision

### DISCUSSION THREAD

**Discussion by j0j003** [ssenart/PyGazpar#48] [Q&A] [answered]
Bonjour,

Tout d'abord merci pour le travail réalisé !

J'ai une question relative à la précision des données. Sur l'exemple que vous donnez on voit que les indexes de début et fin sont en valeur entière, la valeur en kwh est tronqué à sa valeur entière. Pour obtenir une précision on il faut se référé à la valeur volume en m3 :

- start_index_m3: 13695.0,
- end_index_m3: 13702.0,
- volume_m3: 7.7,
- energy_kwh: 86.0,

Là ou cela devient problématique c'est lorsque je regarde mes relevés GrDF à partir d'un export Excel : 

- start_index_m3: 1539,
- end_index_m3: 1539,
- volume_m3: 0.0,
- energy_kwh: 2.0,
- converter_factor_kwh/m3: 10,89

Le volume et les kwh sont clairement tronqués à leur partie entière. Pour obtenir sa consommation (approximative) il faut la calculer à partir du facteur de conversion : 2/10.89 = 0.183m3
Sauf que, si la consommation est (dans le pire des cas) de 2.99 on a réellement consommé 0.274m3 ...

Au départ je pensais à un problème de cast dans l'API ou mon programme mais le problème viendrait de la source...

Êtes vous dans la même situation ou bien ? et deuxième question, les relevés sont censés être journalier et pourtant j'ai un décalage de 3-4jours (nous sommes le 26, dernier relevé le 23), suis-je le seul dans ce cas ?

Bonne journée

**Comment by ssenart** [comment, marked as answer]
Bonjour,

Je viens de vérifier mes relevés récents et, comme vous le dites, les volumes et les énergies consommées sont tronqués à la partie entière. Mon exemple n'est donc pas représentatif des données actuelles.

Pour ma part, je n'utilise que le end_index_m3 et le coefficient de conversion. C'est ce qui correspond vraiment au relevé de mon compteur à la date indiquée. Je recalcule le reste en utilisant le coefficient de conversion. Par exemple, dans Home Assistant, je décris le template correspondant ici: https://github.com/ssenart/home-assistant-gazpar (Home Assistant Energy module integration: Step 1). Les erreurs due à l'arrondie se compensent alors d'un jour sur l'autre (si un jour la mesure me remonte une sous-consommation, le jour suivant j'aurais une sur-consommation d'autant).

Oui, j'ai aussi, comme vous, plusieurs jours de décalage entre le relevé et sa disponibilité sur le site de GrDF (actuellement 3 à 4 jours).

Bien à vous
